distributed/src/test/scala/asobu/distributed/gateway/EndpointSpec.scala

```diff
@@ -0,0 +1,89 @@
+package asobu.distributed.gateway
+
+import akka.actor.{Address, RootActorPath, ActorPath, ActorRef}
+import asobu.distributed.{EndpointDefinition, NullaryEndpointDefinition}
+import asobu.distributed.gateway.Endpoint.Prefix
+import asobu.distributed.service.EndpointDefinitionParser
+import org.specs2.mock.Mockito
+import play.api.mvc.RequestHeader
+import play.api.test.{FakeRequest, PlaySpecification}
+import play.core.routing.RouteParams
+
+import play.routes.compiler._
+import shapeless.HNil
+
+object EndpointSpec extends PlaySpecification with Mockito {
+  import play.api.http.HttpVerbs._
+
+  val routeString =
+    """
+      |# Some Comments
+      |# Some other Comments
+      |GET   /ep1/a/:n   a.b.c(n: Int)
+      |
+      |GET   /ep2/:a/something   a.b.d(a: Int, b: String)
+      |
+    """.stripMargin
+
+  val mockActorRef = mock[ActorRef].defaultReturn(RootActorPath(Address("", "")))
+
+  val createEndpointDef = (route: Route, prefix: Prefix) ⇒ {
+    NullaryEndpointDefinition(prefix, route, mockActorRef): EndpointDefinition
+  }
+
+  lazy val parserResult = EndpointDefinitionParser.parse(Prefix("/"), routeString, createEndpointDef)
+  lazy val endPoints = parserResult.right.get
+  lazy val ep1: EndpointDefinition = endPoints(0)
+  lazy val ep2: EndpointDefinition = endPoints(1)
+
+  "Parse to endpoints" >> {
+    parserResult must beRight[List[EndpointDefinition]]
+    parserResult.right.get.length === 2
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/iHeartRadio/asobu/file/2277672463/issues/source?bid=3037259&fileBranchId=3178877#l41)